sd-dev/bin/status.py

```diff
@@ -72,50 +67,90 @@ def commit_status(status, sha, log):
         status = "pending"
     data["state"] = status
 
-    requests.post(f"https://api.github.com/repos/freedomofpress/securedrop-workstation/statuses/{sha}", json=data, headers=headers)
+    commit_sha = context.get("commit")
+
+    requests.post(f"https://api.github.com/repos/freedomofpress/securedrop-workstation/statuses/{commit_sha}", json=data, headers=headers)
 
 
-def notify_slack(status, log):
+def notify_slack(status, log, context):
     """
     Notifies Slack if the build failed somehow.
     """
     with open("/home/user/.slack-webhook.txt", "r") as s:
         slack_webhook_url = s.readline().strip()
 
+    # Read context in from file
+    commit_message = context.get("message")
+    commit_sha = context.get("commit")
+    commit_author = context.get("author")
+    reason = context.get("reason")
+
+    commit_url = f"https://github.com/freedomofpress/securedrop-workstation/commit/{commit_sha}"
+    if reason == "nightly":
+        text = "This CI run was a nightly automated test of the HEAD commit"
+    elif reason == "push":
+        text = "This CI run was triggered by"
+    else:
+        text = reason
+    text = text + f": <{commit_url}|{commit_sha} by {commit_author}: {commit_message}>"
```

Comment:
I am assuming that the gitpython code in nightlies.py (`message = commit.message`) is the full commit message, so I'm not sure how this will end up displaying for long messages and if we only want to display the first line, e.g. `commit_message.split('\n')[0]`